motion_odometry:
  x_forward_scaling: {
    type: double,
    default_value: 1.25,
    description: "Scaling factor in forward x direction",
    validation: {
      bounds<>: [0.5, 1.5]
    }
  }

  x_backward_scaling: {
    type: double,
    default_value: 0.95,
    description: "Scaling factor in backward x direction",
    validation: {
      bounds<>: [0.5, 1.5]
    }
  }

  y_scaling: {
    type: double,
    default_value: 1.0,
    description: "Scaling factor in y direction",
    validation: {
      bounds<>: [0.5, 1.5]
    }
  }

  yaw_scaling: {
    type: double,
    default_value: 0.6,
    description: "Scaling factor for rotation in yaw direction",
    validation: {
      bounds<>: [0.2, 2.5]
    }
  }

  publish_walk_odom_tf: {
    type: bool,
    default_value: false,
    description:  "Should odom tf be published"
  }

  k: {
    type: double,
    default_value: 0.97,
    description: "Factor for lowpass filtering the foot pressure",
    validation: {
      bounds<>: [0.0, 1.0]
    }
  }

  foot_pressure_topic_left: {
    type: string,
    default_value: "foot_pressure_left/raw",
    description: "Topic for the left foot pressure sensor"
  }

  foot_pressure_topic_right: {
    type: string,
    default_value: "foot_pressure_right/raw",
    description: "Topic for the right foot pressure sensor"
  }


  pressure_vector_size: {
    type: int,
    default_value: 60,
    description: "Size of the buffer for the pressure sensor values which are used to calculate the minima",
    validation: {
      bounds<>: [10, 100]
    }
  }
